editor/utils/with-history/index.js

```diff
@@ -63,6 +99,14 @@ export default function withHistory( reducer, options = {} ) {
 			return state;
 		}
 
+		if ( shouldOverwriteState( action, previousAction ) ) {
```

Comment:
Maybe we can consolidate the return paths by assigning `past` into a variable:

```js
let nextPast = past;
if ( ! shouldOverwriteState( action, previousAction ) ) {
	nextPast = [ ...nextPast, present ];
}

return {
	past: nextPast,
	present: nextPresent,
	future: [],
};
```

Reply:
What would the benefit be? Does it read clearer to you?

Reply:
I personally find fewer return paths tends to be more clear, yes.